BUG: make precommit-check fails on macOS

**Describe the bug**

```
/bin/sh: /opt/miniconda3/bin/pre-commit: Argument list too long
make[1]: *** [precommit-check] Error 126
make: *** [_run] Error 2
```

**ARG_MAX**

```
(base) ➜  semantic-router git:(main) export PRECOMMIT_CONTAINER=ghcr.io/vllm-project/semantic-router/precommit:latest
 docker run --rm -it \
     -v $(pwd):/app \
     -w /app \
     --name precommit-container ${PRECOMMIT_CONTAINER} \
     bash
root@a919dd12c87a:/app# getconf ARG_MAX
2097152
root@a919dd12c87a:/app#


(base) ➜  semantic-router git:(main) getconf ARG_MAX
1048576
```

**Expected behavior**

Without changing macOS default value of `ARG_MAX`, make precommit-check should be successful.

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Additional context**
Add any other context about the problem here.

--- a/tools/make/pre-commit.mk
+++ b/tools/make/pre-commit.mk
@@ -9,12 +9,14 @@ precommit-check: ## Run pre-commit checks on all relevant files
 	@FILES=$$(find . -type f \( -name "*.go" -o -name "*.rs" -o -name "*.py" -o -name "*.js" -o -name "*.md" -o -name "*.yaml" -o -name "*.yml" \) \
 		! -path "./target/*" \
 		! -path "./candle-binding/target/*" \
-		! -path "./.git/*" \
-		! -path "./node_modules/*" \
-		! -path "./vendor/*" \
+    ! -path "./website/node_modules/*" \
+    ! -path "./frontend/node_modules/*" \
+    ! -path "./website/.docusaurus/*" \
 		! -path "./__pycache__/*" \
-		! -path "./site/*" \
+    ! -path "./.venv/*" \
+    ! -path "./vendor/*" \
 		! -name "*.pb.go" \
+    ! -path "./.git/*" \
 		| tr '\n' ' '); \
 	if [ -n "$$FILES" ]; then \
 		echo "Running pre-commit on files: $$FILES"; \

